core/src/main/java/org/mule/DefaultMuleEvent.java

```diff
@@ -605,7 +605,13 @@ public MuleMessage getMessage()
         {
             throw new TransformerException(CoreMessages.objectIsNull("outputType"));
         }
-        return getMuleContext().getTransformationService().getPayload(message, outputType);
+
+        MuleMessage transformedMessage = getMuleContext().getTransformationService().transform(message, outputType);
+        if (ClassUtils.isConsumable(message.getPayload().getClass()))
```

Comment:
We moved the code from DefaultMuleMessage to ClassUtils, that is correct. I'm asking if, from API point of view, makes sense to have MuleMessage#isConsumable. We can discuss it later